### PR TITLE
setup-buildengine.sh: Add --target-basename options

### PR DESCRIPTION
**Pull request by etaishev**
Add base name command line options for configure target and tooling name.

### CODE REVIEW

File: setup-buildengine.sh [hunk 1]
```diff
@@ -392,12 +391,20 @@ while [[ ${1:-} ]]; do
         unregister ) shift
             OPT_DO_UNREGISTER=1
             ;;
+        --basename ) shift
+            OPT_DEFAULT_BASENAME=$1; shift
```
Comment (martyone) [inline setup-buildengine.sh]: I would prefer using "--target-basename" and OPT_TARGET_BASENAME

File: setup-buildengine.sh [hunk 2]
```diff
@@ -392,12 +391,20 @@ while [[ ${1:-} ]]; do
...
+            ;;
         * )
             usage quit
             ;;
     esac
 done
 
+SAILFISH_DEFAULT_TOOLING="${OPT_DEFAULT_BASENAME}-RELEASE"
+SAILFISH_DEFAULT_TARGETS="${OPT_DEFAULT_BASENAME}-RELEASE-i486
+                          ${OPT_DEFAULT_BASENAME}-RELEASE-armv7hl"
```
Comment (martyone) [inline setup-buildengine.sh]: I would prefer keeping the template near the beginning of the file, using "BASENAME" there and replacing it at this place with $OPT_TARGET_BASENAME, similarly to how RELEASE is replaced with $OPT_RELEASE

File: setup-buildengine.sh [hunk 3]
```diff
@@ -301,6 +299,7 @@ Options:
    -vm  | --vm-name <NAME>     create VM with <NAME> [$OPT_VM]
    --no-meta                   suppress creating meta data files with
                                'make-archive-meta.sh'
+    --basename <BASENAME>      basename for the result to targets and kits named as e.g. ${DEF_TARGET_BASENAME}-<version>-<arch>
```
Comment (martyone) [inline setup-buildengine.sh]: I would prefer: "base name for tooling and targets, must use alphanumeric characters only [$OPT_TARGET_BASENAME]" and wrapping the long line. The effect of the option will be available for review before it asks "Do you want to continue?", so I think it is not necessary to expand more on this here.

**Comment by martyone** [comment]
@etaishev I would also consider maintaining your own branch with defaults.sh modified as an alternative to using command line options with values that rarely change.

**Comment by martyone** [comment]
@etaishev please fix commit message and PR title after the changes, I will merge then

**Comment by martyone** [comment]
@etaishev there is still a bit of redundancy in the titles, "setup-buildengine.sh: Add --target-basename option" would be good.

**Comment by martyone** [comment]
Please rebase onto current master.

I am sorry I never noticed your latest update until now - github does not send notifications for these - and so I did not merge it earlier. Better leave a comment whenever you push an update.

**Comment by etaishev** [comment]
@martyone Sorry I missed your comment. Update on master now

**Pull request merged**